### ISSUE TITLE
sql not generated correctly when there is no join table for edge

### ISSUE THREAD

**Issue by RaniRaven** [cytosm/cytosm#23]
for example, using the following gtop :
{
"version" : 1.0,
"abstractionLevel" : {
    "abstractionNodes" : [ {
      "types" : [ "dragon" ],
      "attributes" : [ "id" ]
    }, {
      "types" : [ "guild" ],
      "attributes" : [ "id" ]
    }, {
      "types" : [ "kingdom" ],
      "attributes" : [ "id" ]
    } ],
    "abstractionEdges" : [ {
      "types" : [ "registered_in" ],
      "attributes" : [ ],
      "sourceType" : [ "guild" ],
      "destinationType" : [ "kingdom" ],
      "directed" : true
    }, {
      "types" : [ "owns" ],
      "attributes" : [ "since", "till" ],
      "sourceType" : [ "guild" ],
      "destinationType" : [ "dragon" ],
      "directed" : true
    }, {
      "types" : [ "fires_at" ],
      "attributes" : [ "time" ],
      "sourceType" : [ "dragon" ],
      "destinationType" : [ "dragon" ],
      "directed" : true
    } ]
  },
"implementationLevel" : {
    "graphMetadata" : {
      "backendSystem" : "RELATIONAL"
    },
    "implementationNodes" : [ {
      "types" : [ "dragon" ],
      "tableName" : "dbo.dragon_owners",
      "id" : [ {
        "columnName" : "dragon",
        "datatype" : "VARCHAR(100)",
        "concatenationPosition" : 1
      } ],
      "attributes" : [ {
        "columnName" : "dragon",
        "dataType" : "VARCHAR(100)",
        "abstractionLevelName" : "id"
      } ],
      "restrictions" : [ ]
    }, {
      "types" : [ "guild" ],
      "tableName" : "dbo.guild_in_kingdom",
      "id" : [ {
        "columnName" : "guild",
        "datatype" : "VARCHAR(100)",
        "concatenationPosition" : 1
      } ],
      "attributes" : [ {
        "columnName" : "guild",
        "dataType" : "VARCHAR(100)",
        "abstractionLevelName" : "id"
      } ],
      "restrictions" : [ ]
    }, {
      "types" : [ "kingdom" ],
      "tableName" : "dbo.guild_in_kingdom",
      "id" : [ {
        "columnName" : "kingdom",
        "datatype" : "VARCHAR(100)",
        "concatenationPosition" : 1
      } ],
      "attributes" : [ {
        "columnName" : "kingdom",
        "dataType" : "VARCHAR(100)",
        "abstractionLevelName" : "id"
      } ],
      "restrictions" : [ ]
    } ],
    "implementationEdges" : [ {
      "types" : [ "registered_in" ],
      "paths" : [ {
        "traversalHops" : [ {
          "sourceTableName" : "dbo.guild_in_kingdom",
          "sourceTableColumn" : "guild",
          "destinationTableColumn" : "kingdom",
          "destinationTableName" : "dbo.guild_in_kingdom",
          "attributes" : [ ]
        } ]
      } ]
    }, {
      "types" : [ "owns" ],
      "paths" : [ {
        "traversalHops" : [ {
          "sourceTableName" : "",
          "sourceTableColumn" : "",
          "joinTableSourceColumn" : "guild",
          "joinTableName" : "dbo.dragon_owners",
          "joinTableDestinationColumn" : "dragon",
          "destinationTableColumn" : "",
          "destinationTableName" : "",
          "attributes" : [ {
            "columnName" : "since",
            "dataType" : "DATE",
            "abstractionLevelName" : "since"
          } , {
            "columnName" : "till",
            "dataType" : "DATE",
            "abstractionLevelName" : "till"
          }]
        } ]
      } ]
    }, {
      "types" : [ "fires_at" ],
      "paths" : [ {
        "traversalHops" : [ {
          "sourceTableName" : "",
          "sourceTableColumn" : "",
          "joinTableSourceColumn" : "shooter",
          "joinTableName" : "fires",
          "joinTableDestinationColumn" : "shot",
          "destinationTableColumn" : "",
          "destinationTableName" : "",
          "attributes" : [ {
            "columnName" : "time",
            "dataType" : "DATE",
            "abstractionLevelName" : "time"
          } ]
        } ]
      } ]
    } ]
  }
}

with the query :
MATCH (a :guild)-[:registered_in]->(k :kingdom) RETURN k.id,a.id;

yields -
WITH  SUB_0 AS (
SELECT __src2.guild AS __cytosm9$10_guild, __src2.id AS __cytosm9$10_id, __src3.id AS __cytosm37$38_id, __src3.kingdom AS __cytosm37$38_kingdom
FROM dbo.guild_in_kingdom AS __src2
 JOIN  AS __src5 ON ((__src2.guild = __src5.))
 JOIN dbo.guild_in_kingdom AS __src3 ON ((__src3.kingdom = __src5.))

)
SELECT __src4.__cytosm37$38_id AS "k.id", __src4.__cytosm9$10_id AS "a.id"
FROM SUB_0 AS __src4

If I do put the same table both as join table, I get too many joins which are not being reduced.
According to the documentation of the GTop setting, it should be possible to use implementationNodes without specifying join table

**Comment by rajasaur** [comment]
yes it looks like it skips the join table only if the source and destination are the same.. Got bit by the same problem today and found this during debugging... 

Any chance of a patch anywhere available..
